go/apps/api/routes/v2_keys_verify_key/handler.go

```diff
@@ -0,0 +1,261 @@
+package handler
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/unkeyed/unkey/go/apps/api/openapi"
+
+	"github.com/unkeyed/unkey/go/internal/services/auditlogs"
+	"github.com/unkeyed/unkey/go/internal/services/keys"
+
+	"github.com/unkeyed/unkey/go/pkg/clickhouse"
+	"github.com/unkeyed/unkey/go/pkg/codes"
+	"github.com/unkeyed/unkey/go/pkg/db"
+	"github.com/unkeyed/unkey/go/pkg/fault"
+	"github.com/unkeyed/unkey/go/pkg/otel/logging"
+	"github.com/unkeyed/unkey/go/pkg/ptr"
+	"github.com/unkeyed/unkey/go/pkg/rbac"
+	"github.com/unkeyed/unkey/go/pkg/zen"
+)
+
+type Request = openapi.V2KeysVerifyKeyRequestBody
+type Response = openapi.V2KeysVerifyKeyResponseBody
+
+const DefaultCost = 1
+
+// Handler implements zen.Route interface for the v2 keys.verify endpoint
+type Handler struct {
+	Logger     logging.Logger
+	DB         db.Database
+	Keys       keys.KeyService
+	Auditlogs  auditlogs.AuditLogService
+	ClickHouse clickhouse.ClickHouse
+}
+
+// Method returns the HTTP method this route responds to
+func (h *Handler) Method() string {
+	return "POST"
+}
+
+// Path returns the URL path pattern this route matches
+func (h *Handler) Path() string {
+	return "/v2/keys.verifyKey"
+}
+
+func (h *Handler) Handle(ctx context.Context, s *zen.Session) error {
+	h.Logger.Debug("handling request", "requestId", s.RequestID(), "path", "/v2/keys.verifyKey")
+
+	// Authentication
+	auth, err := h.Keys.GetRootKey(ctx, s)
+	if err != nil {
+		return err
+	}
+
+	// Request validation
+	req, err := zen.BindBody[Request](s)
+	if err != nil {
+		return err
+	}
+
+	res := Response{
+		Meta: openapi.Meta{
+			RequestId: s.RequestID(),
+		},
+		// nolint:exhaustruct
+		Data: openapi.KeysVerifyKeyResponseData{
+			Code:  openapi.NOTFOUND,
+			Valid: false,
+		},
+	}
+
+	key, err := h.Keys.Get(ctx, s, req.Key)
+	if err != nil {
+		return err
+	}
+
+	// Validate key belongs to authorized workspace
+	if key.Key.WorkspaceID != auth.AuthorizedWorkspaceID {
+		return s.JSON(http.StatusOK, res)
+	}
+
+	// Check if API is deleted
+	if key.Key.ApiDeletedAtM.Valid {
+		return s.JSON(http.StatusOK, res)
```

Comment:
nit: personally I'd prefer if we just return full struct here for clarity, it took me a second to understand that this returns a `NOT_FOUND` and I think it's error prone if one isn't careful about it.